flang/include/flang/Frontend/TargetOptions.h

```diff
@@ -0,0 +1,28 @@
+//===--- TargetOptions.h ----------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// Defines the flang::TargetOptions class.
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_FLANG_TARGETOPTIONS_H
+#define LLVM_FLANG_TARGETOPTIONS_H
+
+namespace Fortran::frontend {
+
+/// Options for controlling the target.
+class TargetOptions {
+public:
+  /// The name of the target triple to compile for.
+  std::string triple;
```

Comment:
What else do you plan to add to this `class`?

Reply:
No strict plan just yet. Most likely these:  `hostTriple`, `CPU`, `TuneCPU`. Otherwise - time will tell. See [TargetOptions.h](https://github.com/llvm/llvm-project/blob/d480f968ad8b56d3ee4a6b6df5532d485b0ad01e/clang/include/clang/Basic/TargetOptions.h) from Clang for comparison (it's quite short).

Reply:
OK. Since the class is effectively doing nothing here, you could consider just something like
```c++
/// Target options is just the target triple for now.
using TargetOptions = std::string;
```
in the meantime.

Reply:
I'd like keep this this class as a placeholder and as a hint for various volunteers that will be adding these new features :)

Reply:
I see. It's a weak hint that isn't expressed clearly in the code. There are no comments that the intention is to expand this in any way, etc., which was why I asked about it in the first place.

I think you can add some brief documentation as to the plan, even if somewhat speculative, here.

Reply:
That's a good idea, updated!